### PR TITLE
refactor: fp-ts@2.0

### PR DESCRIPTION
**Pull request by raveclassic**
Hi all!

This PR migrate the codebase to the latest typescript, tslib, tsutils, ts-loader, tslint, remote-data-ts@2.0 and fp-ts@2.0

All other refactoring which is out of scope of migration should be done separately.

### CODE REVIEW

File: packages/react-kit/src/components/DateInput/DateInput.model.ts [hunk 1]
```diff
@@ -100,5 +106,7 @@ export function incrementMonth(month: number): number {
 }
 
 export function incrementMonthOption(month: Option<number>): Option<number> {
-	return month.map(incrementMonth).alt(some(0));
+	return pipe(month, map(incrementMonth), alt(constant(some(0))));
 }
+
+export const inc = (value: number): number => value + 1;
```
Comment (morozovamv) [inline packages/react-kit/src/components/DateInput/DateInput.model.ts]: This function looks like common helper, that doesn't relate to date input model.

File: packages/rx-utils/src/rd/entity-store.utils.ts [hunk 2]
```diff
@@ -34,8 +33,8 @@ export class EntityStore<L = never, A = never> {
 	private isLoadingAll = false;
 	private _getAllValues$ = this.cache.values$.pipe(
```
Comment (morozovamv) [inline packages/rx-utils/src/rd/entity-store.utils.ts]: Why don't you use only one pipe(fp-ts)?

Comment (morozovamv) [reply packages/rx-utils/src/rd/entity-store.utils.ts]: If it is about "other refactoring" could you add issues, please?

File: packages/react-kit/src/components/DateInput/DateInput.model.ts (hunk 1 above)
Comment (raveclassic) [reply packages/react-kit/src/components/DateInput/DateInput.model.ts]: It was moved here from the component. Anyway it is not related to fp-ts@2.0 migration

File: packages/rx-utils/src/rd/entity-store.utils.ts (hunk 2 above)
Comment (raveclassic) [reply packages/rx-utils/src/rd/entity-store.utils.ts]: Unrelated for fp-ts@2 migration.

I'm not sure what issued do you suggest - "refactoring rxjs pipes to fp-ts pipes"? What for?

Comment (morozovamv) [reply packages/rx-utils/src/rd/entity-store.utils.ts]: This is question to you :)
"All other refactoring which is out of scope of migration should be done separately."
What is "All other refactoring"? And should we do this or is it unnecessary?

Comment (raveclassic) [reply packages/rx-utils/src/rd/entity-store.utils.ts]: `All other refactoring` means any refactoring _not_ related to effects of upgrades. This includes: 
 - codestyle
 - deprecations (this case)
 - performance
 - any other code changes

This PR is meant to include minimal changes to keep things working the same as before.

**Pull request merged**